flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -984,6 +989,10 @@ private void connect(Integer headOfChain, StreamEdge edge) {
                         "Data exchange mode " + edge.getExchangeMode() + " is not supported yet.");
         }
 
+        if (resultPartitionType == ResultPartitionType.HYBRID) {
+            hasHybridResultPartition = true;
```

Comment:
We can move this into the `case HYBRID:` branch.

Reply:
If case UNDEFINED reached, it also have chance to set resultPartitionType to HYBRID.

Reply:
Ok, fair enough.